File: ts/src/embed/sample.ts
import { BedrockRuntimeClient, InvokeModelCommand } from '@aws-sdk/client-bedrock-runtime'

const client = new BedrockRuntimeClient({ region: 'us-east-1' })

const fact = "The first moon landing was in 1969."
const animal = "cat"

async function main() {
    const response = await client.send(new InvokeModelCommand({
        body: JSON.stringify({ inputText: fact }),

        //modelId: 'amazon.titan-embed-text-v1',
        modelId: 'amazon.titan-embed-text-v2:0',
        contentType: 'application/json',
        accept: 'application/json',
    }))
    const responseBody = JSON.parse(new TextDecoder().decode(response.body));
    console.log(responseBody.embedding)
}

main();

File: py/src/embed/sample.py
import boto3
import json

client = boto3.client(service_name="bedrock-runtime", region_name="us-east-1")

fact = "The first moon landing was in 1969."
animal = "cat"

response = client.invoke_model(
    body=json.dumps(
        {
            "inputText": animal,
        }
    ),
    modelId="amazon.titan-embed-text-v2:0",
    accept="application/json",
    contentType="application/json",
)

response_body = json.loads(response.get("body").read())
print(response_body.get("embedding"))
